src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/scanner/Scanner.kt

```diff
@@ -34,6 +34,18 @@ interface Scanner : Iterator<Token> {
      */
     fun checkToken(vararg choices: Token.ID): Boolean
 
+    /**
+     * Check if the next token is the given type.
+     *
+     * @param choice token ID to match with
+     * @return `true` if the next token is the given type. Returns `false`
+     * if no more tokens are available.
+     * @throws ScannerException Thrown in case of malformed input.
+     */
+    fun checkToken(choice: Token.ID): Boolean {
```

Comment:
issue: this causes compilation error for Java classes implementing this interface - by default, default methods are not visible in Java. We could use compiler options described [here](https://kotlinlang.org/docs/java-to-kotlin-interop.html#default-methods-in-interfaces) but it might break binary compatibility (we need to be careful here)

Reply:
I wasn't aware of that, thanks! I'll see if migrating all Java inheritors is doable. Probably yes - I cannot check right now, but I think there was just one consumer in the tests.

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/305